File: Problema02/Python/main.py
# Import module math
import math

# Variables
area = perimeter = side_a = side_b = side_c = semi_p = 0

# String format - "{:.4f}".format(variable)
dec_format = "{:.4f}".format

# Inputs
try:
  side_a = float(input("Ingrese primer lado: "))
  side_b = float(input("Ingrese segundo lado: "))
  side_c = float(input("Ingrese tercer lado: "))
except Exception as e:
  print("\nSe produjo la siguiente excepción: " + str(e))

# Process
perimeter = side_a + side_b + side_c
semi_p = perimeter / 2.0
area = math.sqrt(semi_p * (semi_p - side_a) * (semi_p - side_b) * (semi_p - side_c))

# Outputs - F-String Python 3.6
print(f"\nÁrea = {dec_format(area)}")
print(f"Perímetro = {dec_format(perimeter)}")

File: Problema02/Java/Main.java
package io;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.text.DecimalFormat;

public class Main{
   public static void main(String [] args){
       // Declaración de variables y/o constantes + objetos
       BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
       DecimalFormat df = new DecimalFormat(".0000");
       double area, perimeter;
       double sideA = 0, sideB = 0, sideC = 0;
       double semiP;
       
       // Inputs
       try{
           print("\nIngresa primer lado: ");
           sideA = Double.parseDouble(br.readLine());
           
           print("Ingresa segundo lado: ");
           sideB = Double.parseDouble(br.readLine());
           
           print("Ingresa tercer lado: ");
           sideC = Double.parseDouble(br.readLine());
       }catch(Exception e){
           print("\nSe produjo la siguiente excepción: \n" + e.getMessage()); 
       }
       
       // Process
       perimeter = sideA + sideB + sideC;
       semiP = perimeter / 2.0;
       area = Math.sqrt(semiP * (semiP - sideA) * (semiP - sideB) * (semiP - sideC));
       
       // Outputs
       print("\nÁrea = " + df.format(area));
       print("\nPerímetro = " + df.format(perimeter));
   }
   
   public static void print(String s){
     System.out.print(s);  
    }
}
